src/main/java/net/dv8tion/jda/internal/JDAImpl.java

```diff
@@ -664,8 +664,8 @@ public SelfUser getSelfUser()
     public synchronized void shutdownNow()
     {
         shutdown();
-        threadConfig.shutdownNow();
         requester.shutdown(); // stop all requests
+        threadConfig.shutdownNow();
```

Comment:
Since you have changed the order, couldn't `shutdownRequester()` be called instead now or would it make a difference since it is a synchronized method?

Reply:
Calling `shutdownRequester` is unnecessary since we call `shutdownNow` on the executor anyway.

Reply:
As I just realized, you call `threadConfig.shutdownRequester();` in the `shutdownRequester()` method, not `threadConfig.shutdownNow();`. Misread that somehow. My bad.